### PR TITLE
Feature/tg 44 list of classrooms

### PR DESCRIPTION
**Pull request by carlos-canedo-dev**
# Summary

TG-44 - Implementation of list of classrooms

# Details

- Implementation of classrooms controller
- Implementation of classroom service
- Implementation of classroom DTO
- Implementation of http file for classrooms endpoints
- Update of classroom repository with 'GetAllClassroomsAsync' method
- Update of migration script for classrooms seeding
- Unit testing for classroom repository
- Unit testing for classroom service
- Unit testing for classrooms controller

# Evidence
feature-tg-44-list-of-classrooms-be-request
feature-tg-44-list-of-classrooms-be-tests

### CODE REVIEW

File: KlockanAPI/src/KlockanAPI.Application/DTOs/Classroom/CreateClassroomDTO.cs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+﻿namespace KlockanAPI.Application.DTOs.Classroom;
```
Comment (Adrian-Mendoza-Jala) [inline KlockanAPI/src/KlockanAPI.Application/DTOs/Classroom/CreateClassroomDTO.cs:1]: Why empty files?

File: KlockanAPI/src/KlockanAPI.Infrastructure/Data/KlockanContext.cs [hunk 2]
```diff
@@ -172,5 +172,42 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             SessionDuration = 90,
             CreatedAt = new DateTime(2024, 1, 23, 0, 0, 0, DateTimeKind.Utc)
         });
+
+        // Seed data for Program
```
Comment (Adrian-Mendoza-Jala) [inline KlockanAPI/src/KlockanAPI.Infrastructure/Data/KlockanContext.cs]: Update comment

**Review by Adrian-Mendoza-Jala — APPROVED** [review]

File: KlockanAPI/src/KlockanAPI.Presentation/Http/classrooms.http [hunk 3]
```diff
@@ -0,0 +1,7 @@
+@classrooms_url = {{presentation_host}}/api/v1/classrooms
```
Comment (Adrian-Mendoza-Jala) [inline KlockanAPI/src/KlockanAPI.Presentation/Http/classrooms.http:1]: This is for Documentation?

**Review by AJN-jalaU — CHANGES_REQUESTED** [review]
Could we modularize and put the seeders in individual files for each table, inside a seeders folder?

**Review by AJN-jalaU — APPROVED** [review]

File: KlockanAPI/src/KlockanAPI.Application/DTOs/Classroom/CreateClassroomDTO.cs (hunk 1 above)
Comment (carlos-canedo-dev) [reply KlockanAPI/src/KlockanAPI.Application/DTOs/Classroom/CreateClassroomDTO.cs:1]: Actually, the CreateClassroomDTO class is on that file, right now is just a placeholder.

File: KlockanAPI/src/KlockanAPI.Infrastructure/Data/KlockanContext.cs (hunk 2 above)
Comment (carlos-canedo-dev) [reply KlockanAPI/src/KlockanAPI.Infrastructure/Data/KlockanContext.cs]: Fixed in commit [f672744](https://github.com/JU-DEV-LatamBootcamp/Klockan-Backend/commit/f672744c06d72641b52d3612f2253716acd18202).

File: KlockanAPI/src/KlockanAPI.Presentation/Http/classrooms.http (hunk 3 above)
Comment (carlos-canedo-dev) [reply KlockanAPI/src/KlockanAPI.Presentation/Http/classrooms.http:1]: No, those are variables used to execute requests in .http files. I added the REST Client configuration for running those requests in VS Code to the project's README file.

**Pull request merged**